### PR TITLE
DEVTOOLING-82 Task Management Worktype Resource

### PR DESCRIPTION
**Pull request by PrinceMerluza**
Note: On account of bug WORKITEMS-2193. The current implementation includes custom models and custom API call methods for the following:

- Worktype
- Workitemstatus
- Workitemstatuscreate
- Workitemstatusupdate

### CODE REVIEW

**Review by carnellj-genesys — APPROVED** [review]

File: genesyscloud/task_management_worktype/resource_genesyscloud_task_management_worktype.go [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+			}
+			return retry.NonRetryableError(fmt.Errorf("failed to read task management worktype %s: %s", d.Id(), getErr))
+		}
+
+		cc := consistency_checker.NewConsistencyCheck(ctx, d, meta, ResourceTaskManagementWorktype())
+
+		resourcedata.SetNillableValue(d, "name", worktype.Name)
+		resourcedata.SetNillableValue(d, "description", worktype.Description)
+
+		if worktype.Division != nil {
```
Comment (dginty4) [inline genesyscloud/task_management_worktype/resource_genesyscloud_task_management_worktype.go:108]: I think we have a function for readinding divisions

**Review by dginty4 — APPROVED** [review]

**Review by charliecon — APPROVED** [review]

File: genesyscloud/task_management_worktype/genesyscloud_task_management_worktype_proxy.go [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+// deleteTaskManagementWorktypeStatus deletes a Genesys Cloud task management worktype status by Id
+func (p *taskManagementWorktypeProxy) deleteTaskManagementWorktypeStatus(ctx context.Context, worktypeId string, statusId string) (responseCode int, err error) {
+	return p.deleteTaskManagementWorktypeStatusAttr(ctx, p, worktypeId, statusId)
+}
+
+// createTaskManagementWorktypeFn is an implementation function for creating a Genesys Cloud task management worktype
+func createTaskManagementWorktypeFn(ctx context.Context, p *taskManagementWorktypeProxy, taskManagementWorktype *platformclientv2.Worktypecreate) (*platformclientv2.Worktype, error) {
+	log.Printf("Creating task management worktype: %s", *taskManagementWorktype.Name)
+	worktype, resp, err := p.taskManagementApi.PostTaskmanagementWorktypes(*taskManagementWorktype)
+	log.Printf("Completed call to create task management worktype %s with status code %d, correlation id %s and err %s", *taskManagementWorktype.Name, resp.StatusCode, resp.CorrelationID, err)
```
[the remark this answers is not in the document]
Comment (PrinceMerluza) [reply genesyscloud/task_management_worktype/genesyscloud_task_management_worktype_proxy.go:129]: This was recommended by John for debugging purposes. ref: https://github.com/MyPureCloud/terraform-provider-genesyscloud/pull/709#discussion_r1395665925

File: genesyscloud/task_management_worktype/resource_genesyscloud_task_management_worktype.go (hunk 1 above)
Comment (PrinceMerluza) [reply genesyscloud/task_management_worktype/resource_genesyscloud_task_management_worktype.go]: of course! thanks for the catch. fixed.

**Comment by PrinceMerluza** [comment]
Merging with confirmation from @carnellj-genesys

**Pull request merged**